src/shared.ts

```diff
@@ -156,3 +156,7 @@ export const ReflectiveIntrinsicObjectNames = [
     'EvalError',
     'Error',
 ];
+
+export const AsyncFunction = (async () => {}).constructor;
```

Comment:
we are not doing the eval here, which has some weird side effects for some cases, e.g.: if they build the lib with babel, and they try to go down to ES5, it will mess up with the references. This is happening with the `examples/custom-elements/bundle.js`, which is a pitty. I think we will have to use the same destructuring here using eval.